src/components/Calendar/Calendar.js

```diff
@@ -1,185 +1,142 @@
-					const cellMonth = parseInt(cell.getAttribute('data-month'));
-					const cellYear = cell.getAttribute('data-year');
-
-					if (
-						cellDay === eventDay &&
-						(cellMonth + 1).toString() === eventMonth &&
-						cellYear === eventYear
-					) {
-						let visibleEvent = events[i];
-						ReactDOM.hydrate(
-							<CalendarLink event={visibleEvent} cell={cell} />,
-							cell
-						);
-					}
-				}
-			}
-
-			// set month string value
-			document.getElementById('month').innerHTML = months[month];
-			// set year string value
-			document.getElementById('year').innerHTML = year;
-		}
-
-		function daysInMonth(month, year) {
-			// day 0 here returns the last day of the PREVIOUS month
-			return new Date(year, month + 1, 0).getDate();
-		}
-
-		function blankDates(count) {
-			// looping to add the correct amount of blank days to the calendar
-			for (let x = 0; x < count; x++) {
-				let cell = document.createElement('li');
-				let cellText = document.createTextNode('');
-				cell.appendChild(cellText);
-				calendarBody.appendChild(cell);
-			}
-		}
-
-		function next() {
-			year = month === 11 ? year + 1 : year;
-			month = (month + 1) % 12;
-			showCalendar(month, year);
-		}
-
-		function previous() {
-			year = month === 0 ? year - 1 : year;
-			month = month === 0 ? 11 : month - 1;
-			showCalendar(month, year);
-		}
-	};
-	componentDidMount() {
-		this.renderCalendar();
-	}
-
-	render() {
-		return (
-			<div className='events-calendar-container'>
-				<div className='calendar-content'>
-					<h2>Events Calendar</h2>
-					<div id='calendar'>
-						<div className='month'>
-							<ul>
-								<li id='prev'>&#10094;</li>
-								<li id='month'></li>
-								<li id='year'></li>
-								<li id='next'>&#10095;</li>
-							</ul>
-						</div>
-
-						<ul id='weekdays'>
-							<li>Su</li>
-							<li>Mo</li>
-							<li>Tu</li>
-							<li>We</li>
-							<li>Th</li>
-							<li>Fr</li>
-							<li>Sa</li>
-						</ul>
-
-						<ul id='days'></ul>
-					</div>
-				</div>
-			</div>
-		);
-	}
+  constructor(props) {
+    super(props);
+
+
+    const today = new Date();
+    this.state = {
+      dayInt: today.getDate(),
+      month: today.getMonth(),
+      year: today.getFullYear(),
+    };
+  }
+
+  handleClickNext = () => {
+    this.setState(({ month, year }) => {
+      const newYear = year = month === 11 ? year + 1 : year;
+      const newMonth = (month + 1) % 12;
+      
+      return {
+        month: newMonth,
+        year: newYear,
+      };
+    });
+  };
+
+  handleClickPrev = () => {
+    this.setState(({ month, year }) => {
+      const newYear = month === 0 ? year - 1 : year;
+      const newMonth = month === 0 ? 11 : month - 1;
+      
+      return {
+        month: newMonth,
+        year: newYear,
+      };
+    });
+  };
+
+  renderCalendar = () => {
+    const today = new Date();
+    const { events } = this.props;
+    const { dayInt, month, year } = this.state;
```

Comment:
we get the `month` and `year` we want to render from the state - whenever `state` changes, the output of this function changes automatically (because it is used in `render`